packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -214,7 +215,14 @@ class Auth {
 
         this.OCAPISessionsURL = config.OCAPISessionsURL || ''
 
-        this.clientSecret = config.clientSecret || ''
+        // We think there are users of Commerce SDK React and Commerce SDK isomorphic outside of PWA
+        // For these users to use a private client, they must have some way to set a client secret
+        // PWA users should not need to touch this.
+        this.clientSecretPlaceholder = config.enablePrivateClient
+            ? SLAS_SECRET_PLACEHOLDER // PWA proxy is enabled, assume project is PWA
+            : config.clientSecret
+            ? config.clientSecret
+            : ''
```

Comment:
For more readable code:
- Without the chaining, we can simplify it to this.
- How about considering the variable name to be `this.clientSecret`? It's like saying the client secret can be either a placeholder or the real secret.

```suggestion
        this.clientSecret = config.enablePrivateClient
            ? SLAS_SECRET_PLACEHOLDER // PWA proxy is enabled, assume project is PWA
            : config.clientSecret || ''
```

Reply:
For non-pwa-kit users, how will they know to pass in `clientSecret` but _not_ the `enablePrivateClient`? 🤔

Reply:
> For non-pwa-kit users, how will they know to pass in `clientSecret` but _not_ the `enablePrivateClient`? 🤔

Maybe we don't really need the boolean `enablePrivateClient`. We can check the existence of `clientSecret`.

```js
        this.clientSecret = config.clientSecret || SLAS_SECRET_PLACEHOLDER
```

Reply:
> For non-pwa-kit users, how will they know to pass in `clientSecret` but _not_ the `enablePrivateClient`? 🤔

What if they pass in _both_ `enablePrivateClient` and the `clientSecret`? So then our code can be like this:
```js
        this.clientSecret = config.enablePrivateClient
            ? config.clientSecret || SLAS_SECRET_PLACEHOLDER
            : ''
```

Reply:
We've updated this to be the following:
```
        this.clientSecret = config.enablePWAKitPrivateClient
            ? // PWA proxy is enabled, assume project is PWA and that the proxy will handle setting the secret
              // We can pass any truthy value here to satisfy commerce-sdk-isomorphic requirements
              SLAS_SECRET_PLACEHOLDER
            : config.clientSecret || ''
``` 

If `enablePWAKitPrivateClient` is true, it takes precedence over  `clientSecret`. 
We've renamed the first prop to `enablePWAKitPrivateClient` since it is what PWA users will toggle and to make it more clear that non-PWA users should not enable this. 
`clientSecret` is only used by non-PWA Kit projects.